src/main/java/edu/harvard/iq/dataverse/HandlenetServiceBean.java

```diff
@@ -324,22 +322,13 @@ private String getHandleAuthority(String handlePrefix) {
     }
 
     @Override
-    public boolean alreadyExists(Dataset dataset) throws Exception {
-        String handle = getDatasetHandle(dataset);
+    public boolean alreadyExists(DvObject dvObject) throws Exception {
+        String handle = getDvObjectHandle(dvObject);
         return isHandleRegistered(handle);
     }
-
-    @Override
-    public String createIdentifier(Dataset dataset) throws Throwable  {
-        Throwable result = registerNewHandle(dataset);
-        if (result != null)
-            throw result;
-        // TODO get exceptions from under the carpet
-        return getDatasetHandle(dataset);
-    }
-
+    
     @Override
-    public HashMap getIdentifierMetadata(Dataset dataset)  {
+    public HashMap getIdentifierMetadata(DvObject dvObject) {
```

Comment:
I realize this is a bit of an architectural change  now, since this is only available on DOI service beans. Still, would be nice to get rid of this method, or throw a checked exception, or return `Optional<Map>`, to ensure the calling code is ready to handle a case where the `IdServiceBean` it is using cannot get identifier metadata.

Reply:
While we're at it (asking): Why does this return `HashMap` and not `Map<String,String>`?